smdebug/tensorflow/keras.py

```diff
@@ -620,3 +621,75 @@ def new_add_weight(opt, *args, **kwargs):
         # Optimizer is being saved to support additional features in the future.
         self.optimizer = optimizer
         return optimizer
+
+    def forward_pre_hook(self, tape):
+        self.tape = tape
+        self.set_mode(ModeKeys.TRAIN)
```

Comment:
How do we know that it's training mode? Forward pass also occurs in evaluation.

Reply:
Agree, will rely on hook.set_mode called from the training script